src/plopp/graphics/graphicalview.py

```diff
@@ -15,22 +31,112 @@ class GraphicalView(View):
+        cax: Any = None,
+        **kwargs,
+    ):
         super().__init__(*nodes)
-        self.colormapper = None
+        self._dims = dims
+        self._scale = {} if scale is None else scale
+        self.artists = {}
+        self._artist_maker = artist_maker
+        self._kwargs = kwargs
+        self._repr_format = format
+        self._autoscale = autoscale
+        self._bbox = BoundingBox()
+
+        self.canvas = canvas_maker(
+            cbar=cbar,
+            aspect=aspect,
+            grid=grid,
+            figsize=figsize,
+            title=title,
+            vmin=vmin,
+            vmax=vmax,
+            legend=legend,
+            camera=camera,
+            ax=ax,
+            cax=cax,
+        )
+
+        self.colormapper = (
+            ColorMapper(
+                cmap=cmap,
+                cbar=cbar,
+                mask_cmap=mask_cmap,
+                norm=norm,
+                vmin=vmin,
+                vmax=vmax,
+                autoscale=autoscale,
+                canvas=self.canvas,
+                figsize=getattr(self.canvas, "figsize", None),
+            )
+            if colormapper
+            else None
+        )
+
+        if len(self._dims) == 1:
+            self.canvas.yscale = norm
+        self.render()
+
+    def autoscale(self):
+        bbox = BoundingBox()
+        scales = {"xscale": self.canvas.xscale, "yscale": self.canvas.yscale}
+        if hasattr(self.canvas, 'zscale'):
+            scales['zscale'] = self.canvas.zscale
+        for artist in self.artists.values():
+            bbox = bbox.union(artist.bbox(**scales))
+        self._bbox = self._bbox.union(bbox) if self._autoscale == 'grow' else bbox
+        self._bbox = self._bbox.override(self.canvas.bbox)
+        self.canvas.xrange = (
+            _none_if_not_finite(self._bbox.xmin),
+            _none_if_not_finite(self._bbox.xmax),
+        )
+        self.canvas.yrange = (
+            _none_if_not_finite(self._bbox.ymin),
+            _none_if_not_finite(self._bbox.ymax),
+        )
+        if hasattr(self.canvas, 'zrange'):
+            self.canvas.zrange = (
+                _none_if_not_finite(self._bbox.zmin),
+                _none_if_not_finite(self._bbox.zmax),
+            )
+        self.canvas.draw()
 
-    def update(self, *args, **kwargs):
+    def update(self, *args, **kwargs) -> None:
         """
         Update the view with new data by either supplying a dictionary of
         new data or by keyword arguments.
         """
 
         new = dict(*args, **kwargs)
         for key, new_values in new.items():
-            if new_values.ndim != self._ndim:
-                raise ValueError(
-                    f"Expected {self._ndim} dimension(s), but got {new_values.ndim}."
-                )
+            # if new_values.ndim != self._ndim:
+            #     raise ValueError(
+            #         f"Expected {self._ndim} dimension(s), but got {new_values.ndim}."
+            #     )
```

Comment:
Remove or re-enable?

Reply:
So I had disabled this because it no longer made so much sense for the `GraphicalView` to know about the dimensionality of the data. If you supplied data with the wrong number of dimensions, the figure update would still fail, but the error messages would be quite unclear.

I have worked around this in 2 ways:
- In the `GraphicalView`, I check if the supplied data has the coordinates it needs to have, and if not provide info (instead of just a key error)
- In the artists themselves (`Line`, `Image`, `Scatter`, and `Scatter3d`), I perform the dimension check as the artists know what kind of data they are expecting to get (this was already done in the `Scatter3d` and I copied the mechanism over to the others).